tpp-app/src/home/components/CycleHistory.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react';
+import {SafeAreaView, Text, StyleSheet, View, TouchableOpacity, ScrollView} from 'react-native';
+import addDays from 'date-fns/addDays';
+import { STACK_SCREENS } from '../CalendarNavigator';
+import { useNavigation } from '@react-navigation/native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+
+const ShowMore = ({navigation}) => {
+    return(
+        <TouchableOpacity onPress={() => navigation.navigate(STACK_SCREENS.CYCLE_HISTORY, {screen: STACK_SCREENS.CYCLE_HISTORY})}>
+            <View style={styles.showMoreButton}>
+                <Text style={styles.showMoreText}>
+                    Show More
+                </Text>
+                <Icon name="keyboard-arrow-right" size={24} color={"#2F7A6D"}/>
+            </View>
+        </TouchableOpacity>
+    );
+
+}
+
+/**
+ * 
+ * @returns Component that renders 1 continuous period, and lists the start and end
+ */
+function Interval({interval, isMostRecent, onPeriod}){
+    let startDate = interval.start;
+    const formattedStart = startDate.toLocaleString('default', { month: 'short', day: 'numeric' });
+    let days = interval.periodDays;
+    let endDate = addDays(startDate, days);
+    let formattedEnd = endDate.toLocaleString('default', { month: 'short', day: 'numeric' });
+
+    let header = (<View>
+        <Text style={styles.intervalHeader}>{formattedStart} - {formattedEnd}</Text>
+        <Text style={styles.periodText}>{days}-day period</Text>
+    </View>)
+
+    let mostRecentText = onPeriod ? "Current Period" : "Most Recent Period";
+
+    let mostRecentHeader =(
+        <Text style={[styles.intervalHeader, styles.initialHeader]}>
+            {mostRecentText}: Started {formattedStart}
+        </Text>
+    );
+
+    let daysArray = Array.from(Array(days).keys());
+    
+    return (
+        <View style={styles.interval}>
+            {isMostRecent ? mostRecentHeader : header}
+            <View style={{flexDirection: "row"}}>
+                {daysArray.map((day, index) => (
+                    <View style={styles.redDot} key={index}/>
+                ))}
+            </View>
+        </View >
+    )
+}
+/**
+ * 
+ * @returns Component that only renders the 3 most recent periods & their dates in the current year
+ */
+function MinimizedHistoryCard({navigation, intervals, onPeriod}){
+    return (
+        <View style={styles.card}>            
+            <ScrollView style={styles.historyContainer}>
+                <SafeAreaView style={[styles.rowContainer, styles.bottomBorder]}>
+                    <Text style={styles.title}>Period History</Text>
+                    <ShowMore navigation={navigation}/>
+                </SafeAreaView>
+                {intervals.map((interval, index)=> {
+                    if(index < 3){
+                        return  <Interval 
+                                    interval={interval} 
+                                    key={index} index={index} 
```

Comment:
```suggestion
                                    key={index}
```
I don't see the `index` prop being used in `Interval`, perhaps an accidental duplicate?